apps/devhub/views.py

```diff
@@ -628,18 +624,32 @@ def upload(request, addon=None, is_standalone=False, is_listed=True,
     fu = FileUpload.from_post(filedata, filedata.name, filedata.size)
     fu.update(automated_signing=automated)
     log.info('FileUpload created: %s' % fu.pk)
-    if request.user.is_authenticated():
-        fu.user = request.amo_user
+    if user.is_authenticated():
+        fu.user = RequestUser.objects.get(pk=user.pk)
```

Comment:
Do we know what's the RequestUser is used for? It looks like a very (very) thin wrapper around a UserProfile